src/components/Group.js

```diff
@@ -11,39 +11,37 @@ import { FilterIndentTarget } from './DragDrop/IndentTarget'
 import { FilterMoveTarget } from './DragDrop/MoveTargets'
 let { background } = styles
 
-let GroupItem = FilterDragSource(
-  ({
+let GroupItem = FilterDragSource(args => {
+  let {
     child,
     tree,
     index,
     state,
-    parentTree,
     root,
     isRoot,
     connectDragSource,
     //connectDragPreview, isDragging
-  }) =>
-    connectDragSource(
-      <div
-        style={{
-          ...styles.dFlex,
-          ...(index === tree.children.length - 1 &&
-            !root.adding && { background }),
-        }}
-      >
-        {!(isRoot && tree.children.length === 1) && (
-          <Operator
-            {...{ tree, child, root, parentTree, index, parent: state }}
-          />
-        )}
-        {child.join ? (
-          <Group tree={child} root={root} parentTree={tree} />
-        ) : (
-          <Rule node={child} tree={tree} root={root} />
-        )}
-      </div>
-    )
-)
+  } = args
+  return connectDragSource(
+    <div
+      style={{
+        ...styles.dFlex,
+        ...(index === tree.children.length - 1 &&
+          !root.adding && { background }),
+      }}
+    >
+      {!(isRoot && tree.children.length === 1) && (
+        <Operator {...{ ...args, parent: state }} />
+      )}
+      {child.join ? (
+        <Group tree={child} root={root} parentTree={tree} />
```

Comment:
Don’t we need to don’t the ...args stuff here too? Otherwise stuff won’t get propagated down the whole tree

Reply:
You mean in `<Group tree={child} root={root} parentTree={tree} />` or in <Operator?